src/raftstore/store/fsm/peer.rs

```diff
@@ -803,24 +803,23 @@ impl<'a, T: Transport, C: PdClient> PeerFsmDelegate<'a, T, C> {
         }
 
         self.fsm.peer.mut_store().flush_cache_metrics();
-        let res = match res {
+
+        // Keep ticking if there are still pending read requests.
+        if self.fsm.peer.has_unresolved_reads() ||
```

Comment:
Let's keep it inside `check_before_tick` and `check_after_tick` instead of spreading the check everywhere.